src/utility/generateUsers.js

```diff
@@ -0,0 +1,24 @@
+
+function generateUsers(total, hasRating=false) {
+  /*
+  Return an array of user objects.
+  user: {
+    id: uuid
+    username: 
+  }
+  */
+  let users = [];
+  while(total > 0) {
+    let user = {
+      id: total,
+      username: 'user-' + total.toString()
+    };
+    users.push(user);
+    total--;
```

Comment:
a fancy but less performant way to write this (I'm just code golfing)
return [...Array(total).keys()].reverse().map(num => ({id: num, username: `user-${num}`})

As a general guideline and not a strict rule: if a list is > 10-20 items performance trumps readability, if < than that readablity is better.